proposed/http-message.md

```diff
@@ -1265,6 +1265,10 @@ interface UriInterface
      * The path MUST be prefixed with "/"; if not, the implementation MAY
      * provide the prefix itself.
      *
+     * The implementation MUST percent-encode reserved characters as
```

Comment:
- getPath should say that reserved characters MUST be percent-encoded
- withPath should only say that it MUST NOT double-encode characters.

This is because withPath must not encode immediately. That's an implementation detail.